.github/workflows/tox-pytest.yml

```diff
@@ -97,44 +107,46 @@ jobs:
           name: coverage-unit
           path: coverage.xml
 
+  # TODO(rousik): the following could simply run docker
+  # container mounted to specific directory.
+  # TODO(rousik): run this on larger runners, see dev for
+  # how to do that.
   ci-integration:
+    needs:
+      - ci-unit
```

Comment:
Is this dependency required to ensure that the CI shuts down if anything fails, before kicking off the integration tests?

Reply:
This was unfinished attempt to see if we'd like to persist outputs of these as artifacts which could aid analysis, but let's defer this until later.